models/Market.ts

```diff
@@ -0,0 +1,34 @@
+import { FixedProductMarketMaker } from "@/queries/omen";
+import { fromHex } from "viem";
+
+export class MarketModel {
+  data: FixedProductMarketMaker;
+  closingDate: Date;
+  answer: number;
+  isClosed: boolean;
+
+  constructor(market: FixedProductMarketMaker) {
+    this.data = market;
+    this.closingDate = new Date(+market.openingTimestamp * 1000);
+    this.answer =
+      market?.question?.currentAnswer &&
+      fromHex(market.question.currentAnswer, "number");
```

Comment:
Maybe this should be a condition and if there is no `currentAnswer`, answer should be null.

Reply:
👍